src/miniscript/context.rs

```diff
@@ -111,12 +134,31 @@ impl ScriptContext for Segwitv0 {
         frag: &Terminal<Pk, Ctx>,
```

Comment:
If this is only checking a terminal, the function name should have `terminal` rather than `frag` in it.